concourse/steps/replicate_secrets.py

```diff
@@ -11,13 +11,57 @@
 import ccc.secrets_server
 import ci.log
 import ci.util
+import gitutil
 import kube.ctx
 import model
 
 ci.log.configure_default_logging()
 logger = logging.getLogger(__name__)
 
 
+def process_config_queue(
+    cfg_dir: str,
+    repo_url: str,
+    github_repo_path: str,
+    target_ref: str,
+):
+    '''
+    Iterates to be deleted cfg_queue_entries.
```

Comment:
do not repeat the algorithm in the docstr (at least not implementation details such as iteration over elements)

better: find first cfg queue entry that should be deleted, and delete it.